src/trading/tradingSdk.ts

```diff
@@ -23,14 +23,20 @@ interface TradingSdkOptions {
 
 export class TradingSdk {
   constructor(
-    public readonly traderParams: TraderParameters,
+    public traderParams: TraderParameters,
     public readonly options: Partial<TradingSdkOptions> = { enableLogging: false }
   ) {
     if (options.enableLogging) {
       log.enabled = true
     }
   }
 
+  setTraderParams(params: Partial<TraderParameters>) {
+    this.traderParams = { ...this.traderParams, ...params }
+
+    return this
+  }
```

Comment:
This improves it, but I don't know if this would be my approach.

I would probably make the chain part of the calls (at least optional), so you either pass it in the constructor or in the call. 

This doesn't bother me for my PR, as I solved it with a temporal solution (I used a factory method passed in the constructor). After we are done, we might want to do some re-factors in the 2 SDKs or even merge them, so I would not worry too much for now on this yet.

Reply:
Thanks Anxo!
I considered two ways:
1. Extend `params` of each method in order to allow passing traderParams
2. Add one more argument to each method

I did end up with the 1st option, because it's backward compatible and quite short:
https://github.com/cowprotocol/cow-sdk/pull/254/commits/dc1a9c3aa4631dcd169a4ca80144c70081b25c2d

Anyway, if you fill the second way is better - let me know.

Reply:
In line with Anxo's suggestion and your proposal #1, Sasha, would it make sense to make the chainId constructor parameter optional and, if not set in the object and not passed as an optional parameter, throw in the methods?

Reply:
Yes, I was thinking on something in this line. I'd be happy you merged that commit. 

The types and merge logic adds a bit of friction although I got why you did it like this, so if you prefer, to simplify we can remove the trader details from the constructor and make it mandatory in the methods that need it. 

Anyway, I think later we will change a few things in these SDKs, so we can start with https://github.com/cowprotocol/cow-sdk/commit/dc1a9c3aa4631dcd169a4ca80144c70081b25c2d and we'll see later